pgmpy/inference/dbn_inference.py

```diff
@@ -0,0 +1,207 @@
+import itertools
+from collections import defaultdict
+
+from pgmpy.models import BayesianModel, DynamicBayesianNetwork
+from pgmpy.factors import Factor
+from pgmpy.factors.Factor import factor_product
+from pgmpy.inference import Inference, BeliefPropagation
+
+
+class DBNInference(Inference):
+
+    def __init__(self, model):
+        """
+        Class for performing inference using Belief Propagation method
+        for the input Dynamic Bayesian Network.
+        Parameters
+        ----------
+        model: Dynamic Bayesian Network
+        model for which inference is to performed
+        """
+        super().__init__(model)
+        self.interface_nodes_0 = model.get_interface_nodes(0)
+        self.interface_nodes_1 = model.get_interface_nodes(1)
+
+        start_markov_model = self.start_bayesian_model.to_markov_model()
+        one_and_half_markov_model = self.one_and_half_model.to_markov_model()
+
+        combinations_slice_0 = list(itertools.combinations(model.get_interface_nodes(0), 2))
+        combinations_slice_1 = list(itertools.combinations(model.get_interface_nodes(1), 2))
+
+        start_markov_model.add_edges_from(combinations_slice_0)
```

Comment:
`add_edges_from` can accept any iterable so can directly pass the combinations object.